### PR TITLE
Generate Konflux pipelines

### PR DESCRIPTION
**Pull request by pierDipi**
Follow up to https://github.com/openshift-knative/hack/pull/161

- Generate pipeline and pipeline runs for push and PR referencing the Konflux Docker build pipeline (downloaded from https://github.com/konflux-ci/build-definitions/blob/main/pipelines/docker-build-oci-ta/README.md, see updated cmd/konflux-gen/README.md in the PR)

### CODE REVIEW

**Comment by mgencur** [comment]
I haven't fully reviewed this but I can see there's just one type of Pipeline. I was working on https://github.com/openshift-knative/serverless-operator/pull/2707 to build all images there and figured  there are two Pipelines that are slightly different. I call them `serverless-image` and `serverless-index` (the Pipeline for the index image is slightly different). Maybe @creydr could shed some light on the difference as I just copied them from the original pipelines. 
The index image is probably not considered in this PR. Just something to thing about for the future.

**Comment by creydr** [comment]
> I haven't fully reviewed this but I can see there's just one type of Pipeline. I was working on [openshift-knative/serverless-operator#2707](https://github.com/openshift-knative/serverless-operator/pull/2707) to build all images there and figured there are two Pipelines that are slightly different. I call them `serverless-image` and `serverless-index` (the Pipeline for the index image is slightly different). Maybe @creydr could shed some light on the difference as I just copied them from the original pipelines. The index image is probably not considered in this PR. Just something to think about for the future.

The serverless-index pipeline is a bit different, because it has additional steps to verify the file based catalog of the index. This was mostly auto generated, as Konflux recognized it as an index build (therefor I needed to reference a devfile, which I reverted, after we had the pipeline generated (https://github.com/openshift-knative/serverless-operator/pull/2674/commits/2a6573d20e009866cc7b31f094393b1dae6b4b53), as this made things more complicated and was not needed anymore (as we had the FBC related tasks in the pipeline)).

**Comment by pierDipi** [comment]
That pipeline will need to be special cases, it's this one basically https://github.com/konflux-ci/build-definitions/tree/main/pipelines/fbc-builder

**Comment by pierDipi** [comment]
But for now, I'm focusing on dependents component, SO will come next

**Comment by creydr** [comment]
> But for now, I'm focusing on dependents component, SO will come next

ACK. Just wanted to mention it in this context, so we don't forget about it

**Comment by pierDipi** [comment]
Let me know if the PR looks ok and code is clean/readable enough as I'd like to try it out for eventing core

File: pkg/konfluxgen/docker-build-oci-ta.yaml
Comment (creydr) [inline pkg/konfluxgen/docker-build-oci-ta.yaml:1]: Only to make sure I understand this correctly: this is the "default-custom" pipeline from Konflux?

Comment (pierDipi) [reply pkg/konfluxgen/docker-build-oci-ta.yaml:1]: Yes

Comment (pierDipi) [reply pkg/konfluxgen/docker-build-oci-ta.yaml:1]: Actually, it's the new one that is not using PVC for storing artifacts and should solve our issue with Quota

**Review by creydr — COMMENTED** [review]
LGTM from my side

/lgtm

**Pull request merged**